src/main/java/seedu/address/model/Model.java

```diff
@@ -53,22 +54,32 @@
     ReadOnlyAddressBook getAddressBook();
 
     /**
-     * Returns true if a client with the same identity as {@code client} exists in the address book.
+     * Returns true if a client with the same identity as {@code Client} exists in the address book.
      */
     boolean hasClient(Client client);
 
     /**
      * Deletes the given client.
      * The client must exist in the address book.
      */
-    void deleteClient(Client target);
+    void deleteClient(Client client);
 
     /**
      * Adds the given client.
-     * {@code client} must not already exist in the address book.
+     * {@code Client} must not already exist in the address book.
      */
     void addClient(Client client);
 
+    /**
+     * Delete the given Procedure.
+     */
+    void deleteProcedure(Procedure procedure);
+
+    /**
+     * Delete the given Procedure.
+     */
+    void addProcedure(Procedure procedure);
```

Comment:
Same for this method. Connection to which Client?

Reply:
This issue is resolved through telegram: that attribute to query and filter client list is available in the model manager. Hence, this is resolved.